plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -2630,6 +2631,22 @@ public VDI getIsoVDIByURL(final Connection conn, final String vmName, final Stri
         }
     }
 
+    private String actualIsoTemplate(final Connection conn) throws BadServerResponse, XenAPIException, XmlRpcException {
```

Comment:
Would you mind creating a documentation for the method? Then, you do not need comments in line.
Also, this is a good method to create a unit test.

Reply:
I will the documentation, but I really don't think unit testing on this specific method will be possible. Because it needs a live XenServer Host for the method to connect to (both `Host.getByUuid(conn, _host.getUuid());` and `host.getRecord(conn);`) and I'm not sure if they can be Mocked in some way.

Reply:
We do not need a real host. We only want to test the logic. Therefore, you can mock the method calls. PowerMockito and Mockito are there for this kind of job.